### ISSUE TITLE
[Tag] regression: closeIcon not correctly passed through

### ISSUE THREAD

**Issue by mellis481** [ant-design/ant-design#54367]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/react-2ntxvwnz?file=demo.tsx)

### Steps to reproduce

Navigate to reproduction and check `aria-label` of close element.

### What is expected?

It should be what is defined in the code ("Remove This Filter").

### What is actually happening?

Instead, it is always "Close".

| Environment | Info |
| --- | --- |
| antd | 5.26.4 |
| React | Latest |
| System | Windows |
| Browser | Chrome |

---
Up to v5.24.9, you could define the `closeIcon` as the following:
```
<Button
  className="my-button"
  aria-label="Remove This Filter"
  type="primary"
>
  X
</Button>
```
This was required to deal with the accessibility failure caused by the `Tag` component's default close element being just a `span`.

Starting with v25.0.0, a specified `closeIcon` is not being passed through correctly and it always set "Close" as the `aria-label`. This violates another accessibility criterion that programmatic labels should be unique.
